src/flask_pyoidc/pyoidc_facade.py

```diff
@@ -156,79 +166,94 @@ def verify_id_token(self, id_token, auth_request):
         """
         self._client.verify_id_token(id_token, auth_request)
 
-    def refresh_token(self, refresh_token):
-        """
-        Requests new tokens using a refresh token.
+    def refresh_token(self, refresh_token: str):
+        """Requests new tokens using a refresh token.
 
-        Args:
-            refresh_token (str): refresh token issued to client after user authorization
+        Parameters
+        ----------
+        refresh_token: str
+            refresh token issued to client after user authorization.
 
-        Returns:
-            Union[AccessTokenResponse, TokenErrorResponse, None]: The parsed token response, or None if no token
-            request was performed.
+        Returns
+        -------
+        Union[AccessTokenResponse, TokenErrorResponse, None]
+            The parsed token response, or None if no token request was performed.
         """
-        request = {
+        request_args = {
             'grant_type': 'refresh_token',
             'refresh_token': refresh_token,
+            'client_id': self._client.client_id,
+            'client_secret': self._client.client_secret,
             'redirect_uri': self._redirect_uri
         }
+        client_auth_method = self._client.registration_response.get('token_endpoint_auth_method',
+                                                                    'client_secret_basic')
+        return self._client.do_access_token_refresh(request_args=request_args,
+                                                    authn_method=client_auth_method,
+                                                    token=self._token,
+                                                    endpoint=self._client.token_endpoint
+                                                    )
+
+    def _token_request(self, request_args: dict, state: str):
```

Comment:
This entire function should be inlined in `exchange_authorization_code` since it's no longer called from any other place.